spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/StaticGraph.scala

```diff
@@ -46,8 +46,24 @@ class StaticGraph[T: ClassTag](
   private var backId2ForwardId: Array[Int] = _
   private var gradOutputCache: Array[Activity] = _
 
+  compile()
+
   buildBackwardGraph()
 
+  override def getInputShape(): Activity = {
```

Comment:
why not just return inputShapeValue?

Reply:
`inputShapeValue` is just a helper to implement `getInputShape` and it's null for `Graph`. Would not be exposed to user as it's a private field.